### PR TITLE
[WIP] Performance Updates

### PR DESCRIPTION
**Pull request by aulneau**
This PR attempts to address some of the low hanging items related to performance and bundle sizes

Initial efforts prove promising: Performance increases all around, halves the initial time and doubles+ most of the lighthouse scores:

# currently on browser.blockstack.org:
screen shot 2018-08-09 at 3 04 53 pm

## This PR:
screen shot 2018-08-09 at 3 00 07 pm

This PR adds the following:
- React Loadable to help with code-splitting routes
- Lodash plugins for babel/webpack
- A loading state for before js loads, this bumps up our scores considerably:

screen shot 2018-08-09 at 3 37 49 pm

### CODE REVIEW

**Comment by yknl** [comment]
@aulneau can you merge in develop and test this branch when you get a chance? Would be great to have this merged.

File: app/js/routes.js [hunk 1]
```diff
@@ -1,50 +1,232 @@
...
+}
 
-import SignUpPage from './sign-up'
-import SeedPage from './seed'
-import SignInPage from './sign-in'
+const HomeScreenPage = Loadable({
+  loader: () =>
+    import(/* webpackChunkName: "HomeScreenPage" */ './HomeScreenPage'),
+  loading: Loading,
+  delay: 300
```
[the remark this answers is not in the document]
Comment (aulneau) [reply app/js/routes.js]: There are some things we lose related to the webpack/babel output if we make it into a hoc or something like that.

[the remark this answers is not in the document]
Comment (aulneau) [reply app/js/routes.js]: screen shot 2018-08-09 at 3 26 41 pm

[the remark this answers is not in the document]
Comment (wbobeirne) [reply app/js/routes.js]: Just talking about the delay: 300.

[the remark this answers is not in the document]
Comment (aulneau) [reply app/js/routes.js]: oh sure!

File: app/js/routes.js [hunk 2]
```diff
@@ -1,50 +1,232 @@
...
+  if (props.error) {
+    return <div>Error! <button onClick={ props.retry }>Retry</button></div>
+  } else if (props.pastDelay) {
+    return <div style={ {
+      width: '100%',
+      height: '100%',
+      display: 'flex',
+      alignItems: 'center',
+      justifyContent: 'center'
+    } }>Loading...</div>
```
[the remark this answers is not in the document]
Comment (aulneau) [reply app/js/routes.js]: Yeah, I just added another loading state for before js loads, I'll add a spinner or something in both places :)

**Pull request merged**